src/schema/search.ts

```diff
@@ -0,0 +1,155 @@
+import {
+  GraphQLList,
+  GraphQLEnumType,
+  GraphQLString,
+  GraphQLNonNull,
+  GraphQLFieldConfig,
+} from "graphql"
+import { connectionFromArraySlice } from "graphql-relay"
+import { connectionWithCursorInfo } from "schema/fields/pagination"
+import { pageable } from "relay-cursor-paging"
+import { createPageCursors } from "schema/fields/pagination"
+import { SearchableItem } from "./searchableItem"
+
+import { Searchable } from "./searchable"
+import { convertConnectionArgsToGravityArgs } from "lib/helpers"
+
+import { visit, BREAK } from "graphql"
+
+export const SearchIndex = new GraphQLEnumType({
+  name: "SearchIndexes",
+  values: {
+    ARTWORK: {
+      value: "Artwork",
+    },
+    ARTIST: {
+      value: "Artist",
+    },
+  },
+})
+
+export const SearchMode = new GraphQLEnumType({
+  name: "SearchMode",
+  values: {
+    AUTOSUGGEST: {
+      value: "AUTOSUGGEST",
+    },
+    SITE: {
+      value: "SITE",
+    },
+  },
+})
+
+export const searchArgs = pageable({
+  query: {
+    type: new GraphQLNonNull(GraphQLString),
+    description: "Search query to perform. Required.",
+  },
+  indexes: {
+    type: new GraphQLList(SearchIndex),
+    description: "Indexes to include in search. Default: [ARTIST, ARTWORK].",
+  },
+  mode: {
+    type: SearchMode,
+    description: "Mode of search to exceute. Default: SITE.",
+  },
+})
+
+const hydrateSearchResult = (result, root) => {
```

Comment:
An enhancement (not for now! but a nice follow up from @zephraph work), is that we should be batch hydrating results. If we get a list of 10 artists back w/ their partial information from ES, and want to hydrate them, rather than 10 individual fetches for each artist, one batched one would be nice.

I think that's an enhancement and not a blocker b/c we can see if it's even needed (perf might be acceptable w/o the batching enhancement), and if the perf is an issue we can always just skip hydration and use the `SearchableItem` subset of fields (and can even enhance those from Gravity if absolutely needed).